### PR TITLE
Fix query param date bug

### LINKED ISSUE #12228: Issue with date field filters after v0.35.0 upgrade
**Describe the bug**
On a question or a dashboard with a date field filter, there seems to be an issue with deserializing the query parameter. I almost always use relative defaults like the past 30 days, so it may be limited to those strings. Choosing an option from the date filter works, but if you reload the page it fails to parse the query param as below.

**Logs**
I decided to quickly roll back my primary instance once I confirmed this issue, so unfortunately I don't have a full set of logs, but this is the error that is presented in the browser.

`Input to date-string->range does not match schema: [0;33m [(named (not (instance? java.lang.String ["past30days"])) date-string) nil] [0m`

**To Reproduce**
1. Create a native question and/or a dashboard with a field filter against a date column
2. Set the value to something like "past 30 days"
3. Reload the question/dashboard now that the param is set in the URL

**Expected behavior**
Date filter should work, both default values but also when query params are present.

**Screenshots**

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

- Your browser and the version: Chrome, Edge & Safari
- Your operating system: macOS Catalina
- Your databases: BigQuery
- Metabase version: v0.35.0
- Metabase hosting environment: Docker
- Metabase internal database: Postgres

**Severity**
Pretty severe. I usually would try to get more info and submit a better bug report, but in this case I felt I needed to roll back to v0.34.3 right away.

### PR DESCRIPTION
**Pull request by paulrosenzweig**
Fixes #12228 

This fixes the bug, but I'm not sure about the general approach here. We have a handful of components (called "widgets" here) responsible for producing parameter values. However, when they're populated from query params, the values are produced outside of the widgets on the  initial load which can create inconsistencies. Is it worth pushing this responsibility down into whatever component is rendered?

### CODE REVIEW

File: frontend/src/metabase/parameters/components/Parameters.jsx [hunk 1]
```diff
@@ -71,10 +71,14 @@ export default class Parameters extends Component {
         const queryParam = query && query[parameter.slug];
         if (queryParam != null || parameter.default != null) {
           let value = queryParam != null ? queryParam : parameter.default;
-          if (parameter.hasOnlyFieldTargets && value && !Array.isArray(value)) {
-            // FieldValuesWidget always produces an array. If this param has
-            // only field targets we'll use that widget, so we should start with
-            // an array to match.
+
+          // ParameterValueWidget uses FieldValuesWidget if there's no available
```
Comment (tlrobinson) [inline frontend/src/metabase/parameters/components/Parameters.jsx:75]: >  Is it worth pushing this responsibility down into whatever component is rendered?

Yeah, it bothers me that `Parameters.jsx` has special logic relating to specific widgets. If it's not too difficult I think that would be preferable.

**Review by tlrobinson — COMMENTED** [review]
This doesn't make the existing code significantly worse but I don't like that `Parameters.jsx` has logic relating to specific widgets.

**Comment by paulrosenzweig** [comment]
@tlrobinson Sounds good. I probably won't get to that today and don't want this to hold up 0.35.1 and EE. I created [an issue](https://github.com/metabase/metabase/issues/12241) to refactor.

**Comment by mazameli** [comment]
I manually verified this fixes the behavior in #12228 (and the now-closed ##12229).

**Pull request merged**